### ISSUE TITLE
Support PHP 8.0

### ISSUE THREAD

**Issue by phil-davis** [sensiolabs/BehatPageObjectExtension#136]
As at 2020-12-17 support for PHP 8.0 seems to be blocked by dependencies:

- [x] https://github.com/FriendsOfBehat/MinkExtension/pull/4 - looks like the PR is ready now and just needs merge and release. I pinged in the PR today 2021-01-04.

- [x] https://github.com/Ocramius/ProxyManager/pull/658 - merged and release https://github.com/Ocramius/ProxyManager/releases/tag/2.11.0 supports PHP 8.0

- [x] https://github.com/BossaConsulting/phpspec2-expect/pull/65 - I made a PR.  I pinged in the PR today 2021-01-04. Edit: code was refactored so that this dependency is no longer needed.

and???

**Comment by phil-davis** [comment]
PR https://github.com/Ocramius/ProxyManager/pull/658 was merged.

**Comment by jakzal** [comment]
@phil-davis thanks for looking into this!

**Comment by jakzal** [comment]
https://github.com/sensiolabs/BehatPageObjectExtension/pull/137

**Comment by jakzal** [comment]
PHP 8.0 build is now passing: #138

**Comment by jakzal** [comment]
btw, I dropped the expect helper as it seems to be unmaintained.

**Comment by phil-davis** [comment]
https://packagist.org/packages/sensiolabs/behat-page-object-extension does not show the new release yet. Maybe it just needs time, or is some other action needed to push from the 2.3.3 release tag to packagist?

**Comment by jakzal** [comment]
There now :)

**Comment by phil-davis** [comment]
I guess https://github.com/FriendsOfBehat/MinkExtension/pull/4 is not essential for this repo to support PHP 8.0

I got a successful CI run with PHP 8.0 and some UI tests that use Page Objects: 

https://drone.owncloud.com/owncloud/core/28165/124/8
```
8.0: Pulling from owncloudci/php
Digest: sha256:d84b9328af187ddb8592587a3dd8f7ea49947459913480ef335c561a20ccf4b0
Status: Image is up to date for owncloudci/php:8.0
+ make vendor-bin-behat
composer install
...
  - Installing sensiolabs/behat-page-object-extension (v2.3.3): Extracting archive
```

https://drone.owncloud.com/owncloud/core/28165/124/15

```
8.0: Pulling from owncloudci/php
Digest: sha256:d84b9328af187ddb8592587a3dd8f7ea49947459913480ef335c561a20ccf4b0
Status: Image is up to date for owncloudci/php:8.0
...
Running webUIComments tests tagged ~@skipWhenTestingRemoteSystems&&~@skipOnOcV10&&~@skipOnOcV10.6&&~@skipOnOcV10.6.1&&@webUI&&~@skip on browser 'chrome' on platform 'Linux' 
@webUI @insulated @disablePreviews @comments-app-required
Feature: Add, delete and edit comments in files and folders
...
6 scenarios (6 passed)
60 steps (60 passed)
4m41.52s (23.85Mb)
```

Looks good.

**Comment by DonCallisto** [comment]
MinkExtension's PR is merged, so I guess this is ready?

**Comment by DonCallisto** [comment]
There's already a tagged release for this (checked now; didn't noticed), so I guess we can close it?

**Comment by phil-davis** [comment]
https://github.com/FriendsOfBehat/MinkExtension/pull/4 was merged on 2021-01-21 and https://github.com/FriendsOfBehat/MinkExtension/releases/tag/v2.5.0 was released the same day.

https://github.com/sensiolabs/BehatPageObjectExtension/releases/tag/v2.3.3 was released on 2021-01-07. This repo does not use a `composer.lock` so people who use `BehatPageObjectExtension` should get the current `MinkExtension` `2.5.0` when thye do `composer install`

IMO this is finished. Feel free to open if you know otherwise.

**Comment by jakzal** [comment]
thanks both!

> This repo does not use a composer.lock so people who use BehatPageObjectExtension should get the current MinkExtension 2.5.0 when thye do composer install

Just to clarify one thing - even if this repo used the lock file, consumers of this package wouldn’t use it. Composer doesn’t consider lock files of installed packages.
